### DISCUSSION TITLE
Feedback about Windows Installer containing PG, AGE and Graphizer (Graph modeling tool for AGE)

### DISCUSSION THREAD

**Discussion by aked21** [apache/age#1712] [General]
Looking forward to hearing about the Windows installer available at : 
https://age.apache.org/getstarted/quickstart

**Comment by markgomer** [comment]
Software tested in Windows 11 Pro 21H2 build 22000.2538.
The following points were observed in the user experience aspect:

- Windows Smartscreen asked for permission on installing, as the application is not verified. This could bring some concern to firstcomers.
- In order to check the difference between the easy and custom installations, I went back and forth through them. When going back, the “Install” button was not updated to “Next”, so it brought a little bit of confusion if it was really going to install or going to the next page when I clicked it:
![image](https://github.com/apache/age/assets/57829631/5589e6ea-c4be-454a-b9f3-6fe519232cfb)
- While going **back** through the “Easy install” path, the previously hidden steps were displayed.
- The installer detected my existing PostgreSQL 15 installation. 👍
- Went through installing with the "custom" option.
- No shortcut or option to it was created at the installation. As I did not choose the install location, it took me some time finding where to launch the program. But found it's in the file C:\bitnine\graphizer\graphizer.bat.

Aside from that, every other thing worked as expected.

**Reply by markgomer** [reply]
### User Experience Report

* Bad Translations: “Waiting please a few minute” and “Graph data has not created”.

![image](https://github.com/apache/age/assets/57829631/442d3b32-2f86-4714-b539-5581849f3a5e)
![image](https://github.com/apache/age/assets/57829631/8c8639f3-5007-437f-b7be-da8089a4d2fb)

- The process to model the graph is unintuitive. I have followed the tutorial a month ago, but I wasn’t able to repeat it again now. Despite no error being shown in the `preview` window, the software was unable to create the graph data, as shown in the last image.

- Tried to create another node “interests”, but the `preview` window said “interests is not mapped. Mapping is required”, which is very confusing. 

![image](https://github.com/apache/age/assets/57829631/a0b3ba6e-ea7c-44d0-aeb0-b68a4b7ce0fb)

- Turns out I left an empty property in the node. Deleting it made the prior error disappear.

    - This should be automatic, I had given up trying to “map” the node, and wasted a lot of time trying to figure out why it was not mapped.

![image](https://github.com/apache/age/assets/57829631/81e2a979-7994-4585-99d8-77bd891a6faf)

- After adding the interests node, the `error code 500` persisted.

**Comment by JYage** [comment]
There is an issue with the installation stage to address.
Once the installation is about to finish, it brings up the following page, asking if I want to launch Graphizer now after clicking the 'Finish' button.
![1](https://github.com/apache/age/assets/119618967/a7dad57f-0000-42a5-ae1b-753ea1365c0f)

Once I click on the 'Finish' button, the entire installation process seemed to be completed, however, Graphizer was not launched.

I tried the entire installation once again and then finished off by clicking the "Finish" button at the end again.
And then, it showed this pop-up(Image below), which did not appear on my first attempt of the installation.
![image](https://github.com/apache/age/assets/119618967/934f00be-4cff-4781-924d-a70656ba7566)

Upon clicking "Yes", the computer started restarting.
The previous option of "lauch[ing] Graphizer now after clicking the 'Finish' button' does not seem to come in use because the installation requires the computer to restart.

After the restart of the computer, I tried to run Graphizer, which displayed the following page.
![image](https://github.com/apache/age/assets/119618967/409a6143-79fe-46d8-bd9f-cb85335ebbfc)

But Graphizer does not start.
I closed it off and gave it a try multiple times, but it does not work.

**Reply by dpdjvhxm** [reply]
It took about 30 seconds to run the executable file. And the screen in the console window changed, and the web browser was automatically launched to display the Graphizer screen (localhost:3000).

When running in the console window, the image changes to the image below.
![image](https://github.com/apache/age/assets/137974842/717554d2-98d6-4f02-814b-ea36df92ee09)


**※ It may take more time depending on the user's usage environment.**

**Comment by aked21** [comment]
![image](https://github.com/apache/age/assets/76510292/07ffff89-7c5f-4b6b-b8e6-d594b23c75ce)

**After the install, it took almost one minute before Graphizer was launched from the browser.**

**Comment by Garam-Jeon** [comment]
I don’t know if it is my own problem but when I try to install it, the following message comes up.

![screenshot](https://github.com/apache/age/assets/150977365/b6b6d67f-fd82-4f99-bd26-80371b3e83aa)

On the page, there is only don’t run button.

Run away button is only generated when I click on more info.

It will be better if the message does not show up. 

Other than that, it was installed well and quickly.

**Reply by dpdjvhxm** [reply]
This screen is displayed by the security program included in Windows starting with Windows 10 and later versions.

To solve this problem, run it with administrator privileges.

**Comment by dpdjvhxm** [comment]
1. Well I try Apache AGE, I'm not sure about type matter of uppercase and lowercase letters, when I load AGE

![image](https://github.com/apache/age/assets/137974842/4c3f0e0e-43ff-4d53-a308-59c3814cd5d1)


2. I follow the step for AGE Quick start in the website, I'm not sure it is working or not. need some more explanation about result when I type query

![image](https://github.com/apache/age/assets/137974842/f3b20b75-1d3b-4fa1-a51d-1a030bdd417d)

3. About Graphizer, I don't know about what is error code: 500 in there, needs more detail explanation about error-code 

![image](https://github.com/apache/age/assets/137974842/f1c8378b-fcb9-4319-a1d3-f9eda6d6198f)

I'm not sure about much so, If anyone try these, plz share some ideas or suggestions about it.

**Comment by Manji-kw** [comment]
- Using Graphizer, it was good that 3 work places were prepared.


<img width="529" alt="abc1" src="https://github.com/apache/age/assets/111545362/f502f6ee-7f40-4078-9e3d-77ea36b93742">

And also good point was data set was prepared.

I could find tutorial videos at the bottom of main screen 

<img width="532" alt="abc2" src="https://github.com/apache/age/assets/111545362/e09937f4-d258-4902-a3f5-76fe86073f22">



- But I found that I could not change name of node and properties if I copied like below screen shot

<img width="542" alt="abc3" src="https://github.com/apache/age/assets/111545362/c61bbaae-114a-4498-8e08-92da48b1e929">


- Graph modeling was interesting but I really could not find what was the reason of the error. 
- It seems some manual about error numbers caused by which reasons.

<img width="427" alt="abc5" src="https://github.com/apache/age/assets/111545362/e8c76ebe-6051-4ec7-bf13-e7e5ac861c4e">

**Comment by jonaskello** [comment]
Some items encountered during my testing:

This made me unsure if I should install due to security reasons:

- The file would not download because the IP was geo blocked by my firewall (pfBlockerNG). Is it perhaps hosted in China?
- After disabling the firewall I was able to download the zip file. 
- During install got the blue security screen shown in above posts. I could bypass it by pressing the more info link and then run anyway button.

This made me cancel the install:

- Started the installer and selected custom installation
- I only want to install  Graphizer, nothing else since I already have postgres with age running in WSL/Docker.
- There was no option to disable postgres install.
- Cancelled the installation as I don't want to pollute my PC with extra postgres install.

Questions:

Is it possible to only get Graphizer installed without anything else? Do I need to install postgres too in windows even tough I already have it running in docker?

**Comment by markgomer** [comment]
Please refer to this install review on Reddit r/PostgreSQL (REALLY IMPORTANT):
https://www.reddit.com/r/PostgreSQL/comments/1bnv090/comment/kwp6213/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

**Comment by JYage** [comment]
It still does not work on my end, and stays on the same CMD as shown in the image below.


![image](https://github.com/apache/age/assets/119618967/82464a6a-a41b-4c17-b55b-050c41791275)


I let it stay still for almost 15 minutes (or even longer), but it never switched to the web browser.

After noticing that it would not switch to the web browser, I copy-pasted the following address on a web browser:
http://localhost:3000

Then it successfully showed me the following page(as shown in the image below).

![image](https://github.com/apache/age/assets/119618967/d43a17d3-90fa-4c13-baaf-d6d947f64182)

On that screen, once I click on the [Begin Graphizer] button, it displays the following pop-up, saying that "An error occurred while connecting. Please try again later".

![image](https://github.com/apache/age/assets/119618967/c2e82c9d-c048-4a3f-82c4-315edf5631c3)

**Comment by JYage** [comment]
The same issue of being stuck on the CMD has not been solved and is still occurring as shown in the image below.
![image](https://github.com/apache/age/assets/119618967/3df46882-a91f-4e2d-be78-de65cf37cc48)

This time, I let it sit for almost over 20 minutes, and it does not proceed any further.

With the CMD opened, I manually used the following link: http://localhost:3000/login
to open up the browser, which opens up without any technical issues.
But I think that the issue of being stuck on the CMD should be resolved asap.

In addition, even on the browser, which was opened up with the link(http://localhost:3000/login),
the [Begin Graphizer] button still does not work.
It keeps displaying the "An error occurred while connecting. Please try again later."

![image](https://github.com/apache/age/assets/119618967/547d53c4-128c-4648-aeef-61921ec36082)

**Comment by kukim98** [comment]
Hey, did they take down the installer? Seems like it's no longer available for downloads - [Quickstart](https://age.apache.org/getstarted/quickstart/).
